src/lib/components/ChatInput.svelte

```diff
@@ -1,76 +1,302 @@
+    onEnter: onEnter.toString()
   });
 
+  // Action functions for event handling
+  export function handleClick(node: HTMLElement) {
+    const clickHandler = () => {
+      fileInput?.click();
+    };
+
+    node.addEventListener('click', clickHandler);
+
+    return {
+      destroy() {
+        node.removeEventListener('click', clickHandler);
+      }
+    };
+  }
+
+  export function handleChange(node: HTMLElement) {
+    const changeHandler = (event: Event) => {
+      handleFileProcess(event);
+    };
+
+    node.addEventListener('change', changeHandler);
+
+    return {
+      destroy() {
+        node.removeEventListener('change', changeHandler);
+      }
+    };
+  }
+
+  export function handlePaste(node: HTMLElement) {
+    const pasteHandler = async (event: ClipboardEvent) => {
+      // Get clipboard data
+      const items = event.clipboardData?.items;
+      if (!items) return;
+
+      // Check for image data in clipboard
+      for (const item of items) {
+        if (item.type.startsWith('image/')) {
+          // Get the image file from clipboard
+          const file = item.getAsFile();
+          if (!file) continue;
+
+          // Prevent default paste behavior
+          event.preventDefault();
+
+          // Show loading indicator
+          const uploadButton = document.querySelector('[aria-label="Upload image"]');
+          if (uploadButton) {
+            uploadButton.setAttribute('disabled', 'true');
+            uploadButton.setAttribute('title', 'Uploading...');
+          }
+
+          try {
+            // Upload the image using the user.uploadBlob method
+            const uploadResult = await user.uploadBlob(file);
+
+            if (!tiptap) {
+              console.warn("Tiptap editor not initialized");
+              return;
+            }
+
+            // Get the raw image URL without any processing
+            const imageUrl = uploadResult.url;
+
+            // Insert image into editor with the raw URL
+            tiptap.chain().focus().insertContent({
+              type: "image",
+              attrs: { src: imageUrl }
+            }).run();
+
+            // Update content state to ensure persistence
+            content = tiptap.getJSON();
+          } catch (error) {
+            console.error("Error uploading pasted image:", error);
+            toast.error("Failed to upload pasted image", { position: "bottom-right" });
+          } finally {
+            // Re-enable the upload button
+            if (uploadButton) {
+              uploadButton.removeAttribute('disabled');
+              uploadButton.setAttribute('title', 'Upload image');
+            }
+          }
+
+          // Only process the first image found
+          return;
+        }
+      }
+    };
+
+    node.addEventListener('paste', pasteHandler);
+
+    return {
+      destroy() {
+        node.removeEventListener('paste', pasteHandler);
+      }
+    };
+  }
+
+  async function handleFileProcess(event: Event) {
```

Comment:
This function seems to be mostly the same has handlePaste. Especially the try catch. Mind refactoring the duplicate code?